doc/020_installation.rst

```diff
@@ -0,0 +1,76 @@
+############
+Installation
+############
+
+Official Binaries
+*****************
+
+Stable Releases
+===============
+
+You can download the latest stable release versions of rustic from the `rustic
+release page <https://github.com/rustic/rustic/releases/latest>`__. These builds
+are considered stable and releases are made regularly in a controlled manner.
+
+There's both pre-compiled binaries for different platforms as well as the source
+code available for download. Just download and run the one matching your system.
+
+Once downloaded, the official binaries can be updated in place using the 
+``rustic self-update`` command (needs rustic 0.3.1 or later):
+
+.. code-block:: console
+
+    $ rustic  self-update
+    Checking target-arch... x86_64-unknown-linux-musl
+    Checking current version... v0.3.0-dev
+    Checking latest released version... v0.3.1
+    New release found! v0.3.0-dev --> v0.3.1
+    New release is *NOT* compatible
+    
+    rustic release status:
+      * Current exe: "/usr/local/bin/rustic"
+      * New exe release: "rustic-v0.3.1-x86_64-unknown-linux-musl.tar.gz"
+      * New exe download url: "https://api.github.com/repos/rustic/rustic/releases/assets/75146490"
+    
+    The new release will be downloaded/extracted and the existing binary will be replaced.
+    Do you want to continue? [Y/n] Y
+    Downloading...
+    [00:00:00] [========================================] 4.29MiB/4.29MiB (0s) Done
+    Extracting archive... Done
+    Replacing binary file... Done
+    Update status: `0.3.1`!
+
+.. note:: Please be aware that the user executing the ``rustic self-update``
+   command must have the permission to replace the rustic binary.
+
+Unstable Builds
+===============
+
+Another option is to use the latest builds for the master branch, available on
+the `rustic beta download site
+<https://github.com/rustic-rs/rustic-beta>`__. These too are pre-compiled
+and ready to run, and a new version is built every time a push is made to the
+maina branch.
+
+From Source
```

Comment:
Can `cargo install` new used? This would be a good solution, requires no root perms and can be kept updated with `cargo-update`.

Reply:
`cargo install` should work. Don't know about `cargo-update` as I didn't use it so far. 
But also note that there is the `rustic self-update` command which downloads pre-compiled binaries from github.

Reply:
Yes `cargo install rustic-rs` worked (`cargo install rustic` failed because that's a different project 😩).

A self-update function is common on inferior platforms like windows, but on OSes where you have actual package management, using any kind of package manager is superior to self updating each app individually. The latter just does not scale.

E.g. I regularly run this:
```zsh
#!/usr/bin/zsh

set -eux

export PATH="/home/leonie/.cargo/bin:$PATH"

rustup self update
rustup update
cargo install-update --all
cargo cache --autoclean
```

And it updates all my many rust binaries:

```
atuin
cargo-audit
cargo-bloat
cargo-cache
cargo-changelog
cargo-deny
cargo-expand
cargo-generate
cargo-get
cargo-install-update
cargo-install-update-config
cargo-miri
cargo-outdated
cargo-public-api
cargo-release
cargo-sort
cargo-sweep
cargo-tarpaulin
cargo-watch
clippy-driver
delta
dua
grcov
hx
jql
rls
rust-gdb
rust-gdbgui
rust-lldb
rustic
rustup
sk
```

When rustic is packaged by my linux distribution, then I will switch to their well-maintained packages. Until then I must rely on the `cargo-install-update` cargo command supplied by the [`cargo-update`](https://crates.io/crates/cargo-update) crate.

IMHO, if building from _git_ source is documented (which only makes sense if you want to run the latest `main` branch), then building _released versions_ from source and keeping up to date simply via `cargo install` should be documented too. :)